File: externalSystem/app/sanityCheck/sanityCheckController.py
from flask import Blueprint, request
from http import HTTPStatus

from . import SanityCheckService
from .reqDtos.LogsReqDto import LogsReqDto
from .resDtos.LogsResDto import LogsResDto
from .reqDtos.EchoWithJsonReqDto import EchoWithJsonReqDto
from .resDtos.EchoWithJsonResDto import EchoWithJsonResDto

from .._global.logger import CustomLogger
from .._global.logger import CustomLoggerType


bp = Blueprint("sanityCheck", __name__, url_prefix="/sanityCheck")

@bp.route("/", methods=("GET",))
def sanityCheck() -> str :
    CustomLogger.debug(CustomLoggerType.ENTER_EXIT)
    return ("", HTTPStatus.OK)

@bp.route("/logs", methods=("GET",))
def logs() -> str :
    try :

        logsReqDto:LogsReqDto = LogsReqDto(request)


        CustomLogger.debug(CustomLoggerType.ENTER, "", "<logsReqDto: {}>".format(logsReqDto))

        logs = SanityCheckService.logs(logsReqDto)

        CustomLogger.debug(CustomLoggerType.EXIT, "", "<logsReqDto: {}>".format(logsReqDto))
        return (LogsResDto(logs).json(), HTTPStatus.OK)

    except Exception as e :
        CustomLogger.error(e, "", "<lineLength: {}>".format(request.args.get("lineLength") or "None"))
        return ("", HTTPStatus.BAD_REQUEST)

# 정상적인 에러 로그 출력 여부를 테스트해보기 위해서
@bp.route("/divByZeroCheck", methods=("GET",))
def divByZeroCheck() -> str :
    try :

        returnNum:int = 1/0
        return (returnNum, HTTPStatus.OK)

    except Exception as e :
        CustomLogger.error(e, "Div By Zero Check Message", "<returnNum: {}>".format("Undefined"))
        return ("", HTTPStatus.INTERNAL_SERVER_ERROR)

# JSON 송수신 여부를 간편하게 테스트해보기 위해서
@bp.route("/echoWithJson", methods=("PUT",))
def echoWithJson() -> str :
    try :

        echoWithJsonReqDto:EchoWithJsonReqDto = EchoWithJsonReqDto(request)
        CustomLogger.debug(CustomLoggerType.ENTER_EXIT, "", "<echoWithJsonReqDto: {}>".format(echoWithJsonReqDto))
        return (EchoWithJsonResDto(echoWithJsonReqDto.message).json(), HTTPStatus.OK)

    except Exception as e :
        CustomLogger.error(e, "", "<message: {}>".format(request.args.get("message") or "None"))
        return ("", HTTPStatus.BAD_REQUEST)

File: collectedData/src/main/java/toykiwi/sanityCheck/SanityCheckController.java
package toykiwi.sanityCheck;

import toykiwi._global.logger.CustomLogger;
import toykiwi._global.logger.CustomLoggerType;

import toykiwi.sanityCheck.reqDtos.LogsReqDto;
import toykiwi.sanityCheck.reqDtos.MockGeneratedSubtitleUploadedReqDto;
import toykiwi.sanityCheck.reqDtos.MockSubtitleMetadataUploadedReqDto;
import toykiwi.sanityCheck.reqDtos.MockTranlatedSubtitleUploadedReqDto;
import toykiwi.sanityCheck.reqDtos.MockVideoUploadRequestedReqDto;
import toykiwi.sanityCheck.reqDtos.MockVideoUrlUploadedReqDto;
import toykiwi.sanityCheck.resDtos.LogsResDto;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import lombok.RequiredArgsConstructor;

@RestController
@RequiredArgsConstructor
@RequestMapping("/sanityCheck")
public class SanityCheckController {
    private final SanityCheckService sanityCheckService;

    // 정상적인 통신 여부를 단순하게 확인해보기 위해서
    @GetMapping
    public void sanityCheck() {
        CustomLogger.debug(CustomLoggerType.ENTER_EXIT);
        return;
    }

    // 현재 저장된 로그들 중에서 일부분을 간편하게 가져오기 위해서
    @GetMapping("/logs")
    public ResponseEntity<LogsResDto> logs(@ModelAttribute LogsReqDto logsReqDto) {
        try {

            CustomLogger.debug(CustomLoggerType.ENTER);

            LogsResDto logsResDto = this.sanityCheckService.logs(logsReqDto);

            CustomLogger.debug(CustomLoggerType.EXIT, "", String.format("{logsSize: %d}", logsResDto.getLogs().size()));
            return ResponseEntity.ok(logsResDto);

        } catch(Exception e) {
            CustomLogger.error(e, "", String.format("{logsReqDto: %s}", logsReqDto.toString()));
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
        }
    }

    // 정상적인 에러 로그 출력 여부를 테스트해보기 위해서
    @GetMapping("/divByZeroCheck")
    public ResponseEntity<Integer> divByZeroCheck() {
        try {
            Integer returnNum = 1/0;
            return ResponseEntity.ok(returnNum);
        } catch(Exception e) {
            CustomLogger.error(e, "Div By Zero Check Message", String.format("{returnNum: %s}", "Undefined"));
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
        }    
    }

    
    // Policy 테스트용으로 VideoUploadRequested 이벤트를 강제로 발생시키기 위해서
    @PostMapping("/mock/VideoUploadRequested")
    public void mockVideoUploadRequested(@RequestBody MockVideoUploadRequestedReqDto mockData) {
        CustomLogger.debug(CustomLoggerType.ENTER, "", String.format("{mockData: %s}", mockData.toString()));
        this.sanityCheckService.mockVideoUploadRequested(mockData);
        CustomLogger.debug(CustomLoggerType.EXIT);
    }

    // Policy 테스트용으로 VideoUrlUploaded 이벤트를 강제로 발생시키기 위해서
    @PostMapping("/mock/VideoUrlUploaded")
    public void mockVideoUrlUploaded(@RequestBody MockVideoUrlUploadedReqDto mockData) {
        CustomLogger.debug(CustomLoggerType.ENTER, "", String.format("{mockData: %s}", mockData.toString()));
        this.sanityCheckService.mockVideoUrlUploaded(mockData);
        CustomLogger.debug(CustomLoggerType.EXIT);
    }

    // Policy 테스트용으로 SubtitleMetadataUploaded 이벤트를 강제로 발생시키기 위해서
    @PostMapping("/mock/SubtitleMetadataUploaded")
    public void mockSubtitleMetadataUploaded(@RequestBody MockSubtitleMetadataUploadedReqDto mockData) {
        CustomLogger.debug(CustomLoggerType.ENTER, "", String.format("{mockData: %s}", mockData.toString()));
        this.sanityCheckService.mockSubtitleMetadataUploaded(mockData);
        CustomLogger.debug(CustomLoggerType.EXIT);
    }

    // Policy 테스트용으로 GeneratedSubtitleUploaded 이벤트를 강제로 발생시키기 위해서
    @PostMapping("/mock/GeneratedSubtitleUploaded")
    public void mockGeneratedSubtitleUploaded(@RequestBody MockGeneratedSubtitleUploadedReqDto mockData) {
        CustomLogger.debug(CustomLoggerType.ENTER, "", String.format("{mockData: %s}", mockData.toString()));
        this.sanityCheckService.mockGeneratedSubtitleUploaded(mockData);
        CustomLogger.debug(CustomLoggerType.EXIT);
    }

    // Policy 테스트용으로 TranlatedSubtitleUploaded 이벤트를 강제로 발생시키기 위해서
    @PostMapping("/mock/TranlatedSubtitleUploaded")
    public void mockTranlatedSubtitleUploaded(@RequestBody MockTranlatedSubtitleUploadedReqDto mockData) {
        CustomLogger.debug(CustomLoggerType.ENTER, "", String.format("{mockData: %s}", mockData.toString()));
        this.sanityCheckService.mockTranlatedSubtitleUploaded(mockData);
        CustomLogger.debug(CustomLoggerType.EXIT);
    }
}
